.bazelrc

```diff
@@ -74,6 +74,8 @@ build:devdarwinx86_64 --config=dev
 build:dev --stamp --workspace_status_command=./build/bazelutil/stamp.sh
 build:dev --action_env=PATH
 build:dev --host_action_env=PATH
+build:dev --spawn_strategy=local --strategy=Genrule=sandboxed
+build:dev --nostamp
```

Comment:
Pretty weird how the `dev` config defines both `--stamp` and `--nostamp`. Seems wrong. Also, the original issue where the `cockroach` binary is not functional if not stamped doesn't seem to have actually been resolved, has it?

Reply:
Previous discussion at #75976,  pretty sure both things I mentioned in that comment are still true

Reply:
`dev doctor` and running `cockroach start` worked just fine for me, I'm not sure what we were seeing earlier. Was there a specific thing in the CRDB start up process that broke if not stamped? Do you remember the repro steps?

> Pretty weird how the dev config defines both --stamp and --nostamp

Removed the --stamp definition (which got rid of a warning I noticed this branch previously had), but it (i.e. `dev doctor`) still worked BTW.

Reply:
Give me a little bit, I'll investigate and see if I can reproduce the behavior I was seeing before.

Reply:
At the very least you can `rm bin/.dev-status` and re-run `dev doctor` to see it fail. `doctor` has a deliberate test to see if stamping is enabled and you'll break that with this change in the absence of any others.

I haven't been able to reproduce any issue with running the `cockroach` binary when built without stamping which is nice.

Reply:
You gotta be extra careful with this stuff because I think the `stable-status.txt` and `volatile-status.txt` files don't get removed unless you `bazel clean --expunge`. Any testing you do without `clean`ing will pretty much be meaningless.

Reply:
```
$ cockroach git:(220404.skip-sandboxing) bazel clean --expunge
INFO: Invocation ID: 8fbbbb16-e8bb-4182-8ccc-15aa0eb952ba
INFO: Starting clean.

$  cockroach git:(220404.skip-sandboxing) dev doctor
/Users/irfansharif/Software/src/github.com/cockroachdb/cockroach/bin/dev-versions/dev.27 not found, building...
Starting local Bazel server and connecting to it...
INFO: Invocation ID: 039e8ddc-85cc-4819-89df-ffda53fbe6e6
INFO: Analyzed target //pkg/cmd/dev:dev (373 packages loaded, 9954 targets configured).
INFO: Found 1 target...
Target //pkg/cmd/dev:dev up-to-date:
  _bazel/bin/pkg/cmd/dev/dev_/dev
INFO: Elapsed time: 30.276s, Critical Path: 1.75s
INFO: 533 processes: 515 remote cache hit, 16 internal, 2 local.
INFO: Build completed successfully, 533 total actions
INFO: Invocation ID: 05174bc2-933a-4689-9e15-c59ea8e88b8e
You are ready to build :)
```

Reply:
My .bazelrc.user has just the following:
```
build --config=dev
build --config nolintonbuild
```

Could something else be contributing to it working on my machine and not yours? I also re-enabled sandboxing and still the same after `bazel clean --expunge; dev doctor`.

Reply:
Ah, I see what's going on here. You still have `--workspace_status_command` set which is what's causing the `volatile-status.txt` file to get re-created every time, despite `--stamp` not being set. So I do think `dev doctor` will continue to work without any changes, happily.

At the least we can make a tweak to this phrasing in `doctor.go` that is no longer relevant:

    This may be because your Bazel is not configured to "stamp" built executables.

Reply:
Ack, removed it altogether.